[3.13] GH-65961: Document the deprecation of `__package__` and `__cached__` (GH-124377)

miss-islington opened the pull request:
The code changes for warning related to `__package__` landed in Python 3.12. `__cached__` doesn't have any changes as it isn't used but only set by the import system.
(cherry picked from commit 67201ad53ff11576c69a9b762540b77128285f8d)

Co-authored-by: Brett Cannon <brett@python.org>

<!-- gh-issue-number: gh-65961 -->
* Issue: gh-65961
<!-- /gh-issue-number -->


<!-- readthedocs-preview cpython-previews start -->
----
📚 Documentation preview 📚: https://cpython-previews--124380.org.readthedocs.build/

<!-- readthedocs-preview cpython-previews end -->

brettcannon committed:
GH-65961: Document the deprecation of `__package__` and `__cached__` (GH-124377)

The code changes for warning related to `__package__` landed in Python 3.12. `__cached__` doesn't have any changes as it isn't used but only set by the import system.
(cherry picked from commit 67201ad53ff11576c69a9b762540b77128285f8d)

Co-authored-by: Brett Cannon <brett@python.org>

Yhg1s commented on Misc/NEWS.d/next/Core_and_Builtins/2024-09-23-13-25-27.gh-issue-65961.LDqXV2.rst:
```diff
@@ -0,0 +1 @@
+Deprecate the setting and using ``__package__`` and ``__cached__``.
```
```suggestion
Document the deprecation of setting and using ``__package__`` and ``__cached__``.
```

warsaw committed:
Reword news item

Because these attributes were *already* deprecated, the only change now is to document those previous deprecations.

Co-authored-by: T. Wouters <thomas@python.org>

Yhg1s committed:
Merge branch '3.13' into backport-67201ad-3.13
